src/actions/wincher/wincher-keyphrases-action.php

```diff
@@ -0,0 +1,376 @@
+		try {
+			if ( $used_keyphrases === null ) {
+				$used_keyphrases = $this->collect_all_keyphrases();
+			}
+
+			$endpoint = \sprintf(
+				self::KEYPHRASES_URL,
+				$this->options_helper->get( 'wincher_website_id' )
+			);
+
+			$results = $this->client->post(
+				$endpoint,
+				\WPSEO_Utils::format_json_encode(
+					[
+						'keywords' => $used_keyphrases,
+						'url'      => $permalink,
+					]
+				),
+				[
+					'timeout' => 60,
+				]
+			);
+
+			if ( ! \array_key_exists( 'data', $results ) ) {
+				return $this->to_result_object( $results );
+			}
+
+			if ( ! empty( $used_keyphrases ) ) {
+				$results['data'] = $this->filter_results_by_used_keyphrases( $results['data'], $used_keyphrases );
+			}
+
+			// Extract the positional data and assign it to the keyphrase.
+			$results['data'] = \array_combine(
+				\array_column( $results['data'], 'keyword' ),
+				\array_values( $results['data'] )
+			);
+
+			return $this->to_result_object( $results );
+		} catch ( Exception $e ) {
+			return (object) [
+				'error'  => $e->getMessage(),
+				'status' => $e->getCode(),
+			];
+		}
+	}
+
+	/**
+	 * Tracks all the available keyphrases known to Yoast SEO.
+	 *
+	 * @param object $limits The limit API request response.
+	 *
+	 * @return object The tracked keyphrases response object.
+	 */
+	public function track_all( $limits ) {
+		$keyphrases = $this->collect_all_keyphrases();
+
+		if ( empty( $keyphrases ) ) {
+			return $this->to_result_object(
+				[
+					'data'   => [],
+					'status' => 200,
+				]
+			);
+		}
+
+		return $this->track_keyphrases( $keyphrases, $limits );
+	}
+
+	/**
+	 * Collects the keyphrases associated with the post.
+	 *
+	 * @param WP_Post $post The post object.
+	 *
+	 * @return array The keyphrases.
+	 */
+	public function collect_keyphrases_from_post( $post ) {
+		$keyphrases        = [];
+		$primary_keyphrase = $this->indexable_repository
+			->query()
+			->select( 'primary_focus_keyword' )
+			->where( 'object_id', $post->ID )
+			->find_one();
+
+		if ( $primary_keyphrase ) {
+			$keyphrases[] = $primary_keyphrase->primary_focus_keyword;
+		}
+
+		if ( YoastSEO()->helpers->product->is_premium() ) {
+			$additional_keywords = \json_decode( WPSEO_Meta::get_value( 'focuskeywords', $post->ID ), true );
+
+			$keyphrases = \array_merge( $keyphrases, $additional_keywords );
+		}
+
+		return $keyphrases;
+	}
+
+	/**
+	 * Collects all keyphrases known to Yoast.
+	 *
+	 * @return array
+	 */
+	protected function collect_all_keyphrases() {
+		// Collect primary keyphrases first.
+		$keyphrases = \array_column(
+			$this->indexable_repository
+				->query()
+				->select( 'primary_focus_keyword' )
+				->where_not_null( 'primary_focus_keyword' )
+				->find_array(),
+			'primary_focus_keyword'
+		);
+
+		if ( YoastSEO()->helpers->product->is_premium() ) {
+			// Collect all related keyphrases.
+			$query_posts = new WP_Query( 'post_type=any&nopaging=true' );
```

Comment:
We should change this query to query directly on the postmeta table to retrieve all entries there without first retrieving all posts. On larger sites this is bound to give difficulties.